From 3e556a5e375faf01a49d3519d10323e2d98029d2 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:22 -0400
Subject: [virt] kvm/emulator: cleanup decode_register_operand() a bit

Message-id: <1406040016-3289-108-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86186
O-Subject: [RHEL7 PATCH v2 107/221] KVM: emulator: cleanup decode_register_operand() a bit
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Gleb Natapov <gleb@redhat.com>

Make code shorter.

Reviewed-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit 6d4d85ec5698a65165eb9af0ce0e508a7c757bac)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/emulate.c b/arch/x86/kvm/emulate.c
index 4c7b4b9..205b17e 100644
--- a/arch/x86/kvm/emulate.c
+++ b/arch/x86/kvm/emulate.c
@@ -1045,13 +1045,9 @@ static void decode_register_operand(struct x86_emulate_ctxt *ctxt,
  }
 
  op->type = OP_REG;
- if (ctxt->d & ByteOp) {
-  op->addr.reg = decode_register(ctxt, reg, true);
-  op->bytes = 1;
- } else {
-  op->addr.reg = decode_register(ctxt, reg, false);
-  op->bytes = ctxt->op_bytes;
- }
+ op->bytes = (ctxt->d & ByteOp) ? 1 : ctxt->op_bytes;
+ op->addr.reg = decode_register(ctxt, reg, ctxt->d & ByteOp);
+
  fetch_register_operand(op);
  op->orig_val = op->val;
 }
-- 
1.7.1